lib/kernel-manager.coffee

```diff
@@ -149,9 +151,9 @@ module.exports = KernelManager =
 
         kernelSpec.grammarLanguage = grammarLanguage
 
-        ConfigManager.writeConfigFile (filepath, config, onlyConnect = false) =>
-            kernel = new Kernel kernelSpec, grammar, config, filepath, onlyConnect
+        customKernelConnectionPath = path.join atom.project.rootDirectories[0].path, 'hydrogen', 'connection.json'
 
+        finishKernelStartup = (kernel) =>
```

Comment:
I don't think this is necessary.

CoffeeScript is smart enough to handle the scope of `kernel` (even if assigned within a try-catch block).

The CoffeeScript code below:
```coffee
try
    foo = false
catch err
    foo = true

console.log foo
```
compiles to
```javascript
var err, error, foo;

try {
  foo = false;
} catch (error) {
  err = error;
  foo = true;
}

console.log(foo);
```